### DISCUSSION TITLE
In-game currency system

### DISCUSSION THREAD

**Discussion by WoahdangJr142** [ppy/osu#19197] [Ideas]
I think it would be a super cool feature to have a sort of currency system in the game. Preferably not purchasable be real money, as that seems like it would go against the games ideals. I’m thinking it could be earned by things such as reaching pp milestones, leveling up, playing best maps, creating best maps, and modding best maps. Maybe some other things too. I’m thinking they could be used to buy things such as skins (remove skin importing) best maps, and maybe add a couple other things to buy such as name cards, avatars, and titles. Again, these items could easily be changed and don’t have to be exactly what I suggested. I think this would be a very cool feature to add while not taking anything away from the game. Maybe later a trade system could even be added, but that is a different discussion. I didn’t have much time to write this out and I’m on my phone so to clear anything up just ask

**Comment by frenzibyte** [comment]
> I’m thinking they could be used to buy things such as skins (remove skin importing)

You mean the user will have to gain coins in order to buy skins? We're not going a Fortnite route here or something.

I'm not sure I agree with the idea as a whole, the user should have full customisation over their profile to their comfort. This is almost reaching a micro-transaction model without the transaction part of it, and I'm not sure that's fun.

**Reply by WoahdangJr142** [reply]
Oh yeah absolutely I don’t want it to be micro transactions like how you said. That’s why I said you shouldn’t be able to buy the currency. Let me re-phrase a little bit. For the skins and beatmaps there could be larger collections of “free” stuff, and then there could also be some items that you can unlock, and they could range from just being a little fun for pretty cheap or a flex for pretty expensive. And idk what I was thinking when I wrote remove importing. Ig I just should have put make a way to make sure that you can’t export and import the stuff you bought. Does this make a little more sense?

**Reply by frenzibyte** [reply]
Having the concept of "paid" skins isn't really all that easy, recreating them is not that hard, and osu!(lazer) isn't aiming towards encrypting skin files to avoid users exporting them in their own ways.

**Reply by WoahdangJr142** [reply]
Oh that’s true. Maybe not the skins or necessarily maps, but nameplates, banners, titles, and other small things like that

**Reply by nagi-desuuu** [reply]
I think osu would be going against their decision for a free-to-win decision should this gets accepted.
We already have custom cover photos bundled with osu!supporter, and that's enough. Not to mention the me! tab that you get in the website.
